hls4ml/model/optimizer/optimizer.py

```diff
@@ -44,3 +44,13 @@ def optimize_model(model, passes=None):
             break
         else:
             optimization_done = True
+
+def node_output_use_map(model, node):
+    output_map = {}
+    for output in node.outputs:
+        output_map[output] = []
+        for layer in model.get_layers():
+            for inp in layer.inputs:
+                if output == inp:
+                    output_map[output].append(layer)
+    return output_map
```

Comment:
Maybe this should be a member function of `Layer` in `hls_layers.py` ? And at the same time, it would make sense to tidy up `Layer.get_output_nodes` to loop over outputs instead of picking just the first to make it consistent with this.

Reply:
@thesps this is done in a4870f8d3e3ae81a01562141b84e7f91a4c9963d